### ISSUE TITLE
General Description about Working Principles

### ISSUE THREAD

**Issue by vanthome** [lanthaler/HydraConsole#8]
I think we need a short description about the working principles of the console.
In my opinion best told with some use cases like these:

UC1 - GET API Documentaion:
1.) Make a request to the given API endpoint 
2.) Client detects type `ApiDocumentation`
3.) Render interactive representation of the documentaion

UC2 - GET Collection/ Paged Collection:
1.) Make a request to the given API endpoint 
2.) Client detects type `Collection` / `PagedCollection`
3.) Render interactive representation of the collection

UC3 - GET Resource:
1.) Make a request to the given API endpoint 
2.) Client detects type `Resource`
3.) Render interactive representation of the resource

etc.

Also, I propose the following enhancements:
- Refactor Hydra client so that it is independent from Backbone (may later be put into a seperate project)
- Error handling must be better: Notify the user that a response could not be interpreted
- Remove dependency to PHP proxy as the Vocabs should now reside on locations that return proper CORS headers

**Comment by wancharle** [comment]
Ps.: sorry my english: i don't speak english!
I think this is important . 
i was using the console to learning, but its become complicated... i tried mimic the jsons responses of event-api demo, but i have many mistakes... example: i tried put the code bellow to response of entry point on my server: 

```javascript
{
  "@context": "/hydra/event-api/contexts/EntryPoint.jsonld",
  "@id": "/hydra/event-api/",
  "@type": "EntryPoint",
  "events": "/hydra/event-api/events/"
}
```

but i have errors...then i figure out this... the response expected by the console is this:

```javascript
{
    "@context": {
        "__value": "/hydra/event-api/contexts/EntryPoint.jsonld",
        "__activectx": {
            "@base": {},
            "hydra": {
                "@id": "http://www.w3.org/ns/hydra/core#",
                "@reverse": false
            },
            "vocab": {
                "@id": "http://www.markus-lanthaler.com/hydra/event-api/vocab#",
                "@reverse": false
            },
            "EntryPoint": {
                "@id": "http://www.markus-lanthaler.com/hydra/event-api/vocab#EntryPoint",
                "@reverse": false
            },
            "events": {
                "@id": "http://www.markus-lanthaler.com/hydra/event-api/vocab#EntryPoint/events",
                "@reverse": false,
                "@type": "@id"
            }
        }
    },
    "@id": {
        "__iri": "@id",
        "__value": {
            "__orig_value": "/hydra/event-api/",
            "__value": {
                "@id": "http://www.markus-lanthaler.com/hydra/event-api/"
            }
        }
    },
    "@type": {
        "__iri": "@type",
        "__value": {
            "__orig_value": "EntryPoint",
            "__value": {
                "@id": "http://www.markus-lanthaler.com/hydra/event-api/vocab#EntryPoint"
            }
        }
    },
    "events": {
        "__iri": "http://www.markus-lanthaler.com/hydra/event-api/vocab#EntryPoint/events",
        "__value": {
            "__orig_value": "/hydra/event-api/events/",
            "__value": {
                "@id": "http://www.markus-lanthaler.com/hydra/event-api/events/"
            }
        }
    }
}
```

 what is this properties : __value, __orig_value, __activectx etc?  

I suppose it's the result of the recursive processing of the original json.But I do not know how I can do this processing in my api. is a different form of serialization json? I've been researching and found that there are different forms (expanded, flatened, framed) but do not know how the console is waiting ...

**Comment by lanthaler** [comment]
Good point @wancharle. In 7d1b83721236b3d1625c8184e032cdd9d8b058df I added a reference to the [high-level description of how the HydraConsole works](http://m.lanthi.com/3gen-web-apis-p171) in my dissertation. If you have further questions, feel free to join the Hydra W3C Community Group and post your questions directly to our mailing list. All details can be found here: http://www.hydra-cg.com/#community

**Comment by wancharle** [comment]
thanks @lanthaler i will go read it!

**Comment by lukasgraf** [comment]
After searching for the implementation of the extended expansion algorithm for quite a while: It's in the [`expansion-debug`](https://github.com/lanthaler/JsonLD/tree/expansion-debug) branch of the [`JsonLD`](https://github.com/lanthaler/JsonLD) package, starting with [`aa331099`](https://github.com/lanthaler/JsonLD/commit/aa331099d7749700209ddf11e7f11ecfa5d00793).

**Comment by wancharle** [comment]
thank you! this helped me a lot.
